src/App.js

```diff
@@ -2,355 +2,54 @@ import React from "react";
-
-    const digitAndP = /([0123456789])(\()/g;
-    const pAndDigit = /(\))([0123456789])/g;
-    // )( -> )*(
-    const pAndp = /(\))(\()/g;
-
-    //return the value
-    return expression
-      .replace(digitAndP, "$1*$2")
-      .replace(pAndDigit, "$1*$2")
-      .replace(pAndp, "$1*$2");
-
-
-  };
-
-
-
-  calculate = () => {
-    const { result, operate, originalLastNum } = this.state;
-    let finalResult = 0;
-    var tempResult = result;
-
-    // Special case with expression ending with operator, after CE use
-    if ("-+*/".includes(this.getLastChar(result))) {
-      tempResult = result.slice(0, -1);
+      this.setState({
+        result: values[0] + values[1],
+        lastEquation: "",
+      });
     }
 
+    if (button.key === "=") {
+      var finalValues = finalizeEquation(result, lastEquation);
 
-    tempResult = this.closeParens(tempResult) + "";
-    tempResult = this.addMultiplier(tempResult) + "";
-
-
-    // Handling double equal
-    // If I only have sign + digit(s) + dot + digit(s)
-    const regex = /-{0,1}[0123456789]*(\.[0123456789]*){0,1}/g;
-    const matches = tempResult.match(regex) || [];
-    if (matches.length > 1 && matches[0] === tempResult) {
-      //I need to handle the double equal
-      if ("+-/*".includes(operate) && !isNaN(originalLastNum)) {
-        // finalResult = evaluate(result + operate + originalLastNum) + "";
-        try {
-          finalResult = this.roundedResult(tempResult + operate + originalLastNum) + "";
-        } catch(error) {
-
-          //throw error
-        }
-      }
-    } else {
-      var tempResultString = tempResult;
-      //We need to detect the non nested parenthesis
-      //Open ( and no other ( before the next close
-      // '(' 0123456789-+*/.(not (, not ) ) ')'
-      // ( + any number of the 15 chars + ) : (7+2) (-3.25478/+6.587)
-      const reg = /\(([0123456789/*-+.]*)\)/g;
-      var parenthesisToCalculate = tempResultString.match(reg) || [];
-      while (parenthesisToCalculate.length > 0) {
-        for (var i = 0; i < parenthesisToCalculate.length; i++) {
-          //we extract the first expression ex:  (2+5)
-          let expression = parenthesisToCalculate[i];
-          //We calculate the value ex: 7
-
-          try {
-            let tempResult = round(evaluate(expression), 13);
-
-            //We need to replace the expression by the calculation
-            tempResultString = tempResultString.replace(expression, tempResult);
-
-          } catch (error) {
-            //throw error
-          }
-        }
-
-        parenthesisToCalculate = tempResultString.match(reg) || [];
-      }
-      //In tempResultString we have the last expression withtout any ()
-
-      try {
-       finalResult = round(evaluate(tempResultString), 13);
-      } catch (error) {
-        //throw error
-      }
       this.setState({
-        done: true,
-        result: finalResult + "",
+        result: finalValues[1] + "",
```

Comment:
@grimxreaper This works!  But is there a way to have these values as named variables instead of array values?

Reply:
How about returning in an object with labels?

Reply:
That could work too! I think there's a nice es6 feature for turning array values into variables that you could use though...

Reply:
I was also thinking about destructuring, okay I will do that!

Reply:
Done

Reply:
@grimxreaper I'm still seeing `finalresult[1]`

Reply:
I overlooked that one, fixed now, thank you!